docs/src/main/tut/datatypes/validated.md

```diff
@@ -12,7 +12,351 @@ Response comes back saying your username can't have dashes in it, so you make so
+4. The [applicative](../typeclasses/applicative.html) syntax `(a, b, c, ...).mapN(...)` provides us a way to accumulatively apply the validation functions and yield a product with their successful result or the accumulated errors in the `NonEmptyList`. Then, we transform that product with `mapN` into a valid instance of `RegistrationData`.
+
+**Deprecation notice:** since cats `1.0.0-MF` the cartesian syntax `|@|` for applicatives is deprecated. If you're using `0.9.0` or less, you can use the syntax: `(a |@| b |@| ...).map(...)`.
+
+Note that, at the end, we expect to lift the result of the validation functions in a `RegistrationData` instance. If the process fails, we'll get our `NonEmptyList` detailing what went wrong.
+
+For example:
+
+```tut:book
+FormValidatorNel.validateForm(
+  username = "Joe",
+  password = "Passw0r$1234",
+  firstName = "John",
+  lastName = "Doe",
+  age = 21
+)
+
+FormValidatorNel.validateForm(
+  username = "Joe%%%",
+  password = "password",
+  firstName = "John",
+  lastName = "Doe",
+  age = 21
+)
+```
+
+Sweet success! Now you can take your validation process to the next level!
+
+### A short detour
+
+Typically, you'll see that `Validated` will be accompanied by a `NonEmptyList` when it comes to accumulation. The thing here is that you can define your own accumulative data structure and you're not limited to the aforementioned construction.
+
+For doing this, you have to provide a `Semigroup` instance. `NonEmptyList`, by definition has its own `Semigroup`. For those who don't know what a `Semigroup` is, let's see a simple example.
+
+#### Accumulative Structures
+
+According to [Wikipedia](https://en.wikipedia.org/wiki/Semigroup):
+
+> A semigroup is an algebraic structure consisting of a set together with an associative binary operation.
+
+You can find more about how `Semigroup` works in cats [here](../typeclasses/semigroup.html).
+
+Let's take a look at `ap` method of `Validated`:
+
+```tut:silent:fail
+/**
+   * From Apply:
+   * if both the function and this value are Valid, apply the function
+   */
+def ap[EE >: E, B](f: Validated[EE, A => B])(implicit EE: Semigroup[EE]): Validated[EE, B] =
+  (this, f) match {
+    // ...
+    case (Invalid(e1), Invalid(e2)) => Invalid(EE.combine(e2, e1))
+    // ...
+  }
+}
+```
+
+We've omitted the complete implementation because our focus here is the case in where you need to append (that's the function of this method) two failures. Note the `implicit EE: Semigroup[EE]` parameter and the usage of its `.combine` operation. In the case of `NonEmptyList`, we're talking about a `List`, with certain properties that allow us to _combine_ (append) more than one element to it. That's because, apart from the fact that it is a `List`, it also has an instance of `Semigroup`, telling it how to operate with the accumulation.
```

Comment:
I find this section a bit confusing.

Maybe something like : 

> We've omitted the complete implementation and only show the part where two failures are combined using the `combine` method of the [`Semigroup`](https://typelevel.org/cats/typeclasses/semigroup.html) instance of `EE`.

I don't think we need the part about `NonEmptyList`. Maybe you can add an example of `NonEmptyList`'s `Semigroup` to the Accumulative Structures section ?

Something like : 

```scala
NonEmptyList.one("error 1") |+| NonEmptyList("error 2", "error 3")

"error 1".invalidNel[Int] |+| "error 2".invalidNel
("error 1".invalidNel[Int], "error 2".invalidNel[Int]).mapN(_ + _)
```